projects/RabbitMQ.Client/client/impl/WireFormatting.cs

```diff
@@ -310,7 +323,7 @@ public static int ReadTimestamp(ReadOnlySpan<byte> span, out AmqpTimestamp value
 
         public static int WriteArray(Span<byte> span, IList val)
         {
-            if (val is null)
+            if (val is null || val.Count == 0)
```

Comment:
Could we null coalesce here?

Reply:
Yeah, comes out as `(val?.Count ?? 0) == 0`. Not much prettier :P

Reply:
Or `!(val?. Count > 0)`

Reply:
`list is { Count: > 0 }` on C# 9 :)

Reply:
Same pattern exists on various other methods in this class as well. I'd argue the current one is more readable than any other proposed solution

Reply:
Agreed